src/azure-cli/azure/cli/command_modules/acs/managed_cluster_decorator.py

```diff
@@ -5160,6 +5232,39 @@ def set_up_oidc_issuer_profile(self, mc: ManagedCluster) -> ManagedCluster:
 
         return mc
 
+    def set_up_workload_auto_scaler_profile(self, mc: ManagedCluster) -> ManagedCluster:
+        """Set up workload auto-scaler profile for the ManagedCluster object.
+
+        :return: the ManagedCluster object
+        """
+        self._ensure_mc(mc)
+
+        if self.context.get_enable_keda():
+            if mc.workload_auto_scaler_profile is None:
+                mc.workload_auto_scaler_profile = self.models.ManagedClusterWorkloadAutoscalerProfile()
+            mc.workload_auto_scaler_profile.keda = ManagedClusterWorkloadAutoscalerProfileKeda(enabled=True)
+
+        return mc
+
+    def update_workload_auto_scaler_profile(self, mc: ManagedCluster) -> ManagedCluster:
```

Comment:
You should move this function to the other class `AKSManagedClusterUpdateDecorator`. The function to construct the updated profile (`update_mc_profile_default`) is also in this class.